src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,198 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int OFFSET_ONE = 1;
+    private static final int EMPTY_LIST = 0;
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        public Node(Node<E> prev, E item, Node<E> next) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. It seems you are already doing this correctly, but make sure to maintain this practice throughout your code.